synapse/http/federation/well_known_resolver.py

```diff
@@ -76,7 +76,7 @@
 
 @attr.s(slots=True, frozen=True)
 class WellKnownLookupResult:
-    delegated_server = attr.ib()
+    delegated_server: Optional[bytes] = attr.ib()
```

Comment:
Can we `auto_attribs=True`, while we're here? 😄

Reply:
Sure, I'll chuck this in before merging.